lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -365,6 +365,29 @@ def linearized_ancestors_of(fully_qualified_name)
       ancestors
     end
 
+    # Resolves an instance variable name for a given owner name. This method will linearize the ancestors of the owner
+    # and find inherited instance variables as well
+    sig { params(variable_name: String, owner_name: String).returns(T.nilable(T::Array[Entry::InstanceVariable])) }
+    def resolve_instance_variable(variable_name, owner_name)
+      entries = T.cast(self[variable_name], T.nilable(T::Array[Entry::InstanceVariable]))
+      ancestors = linearized_ancestors_of(owner_name)
+      return unless entries && ancestors.any?
```

Comment:
Should we `return unless entries` first before we call `linearized_ancestors_of` to potentially save the computation in some cases?

Reply:
Good point! I'll create another PR.